### DISCUSSION TITLE
Confused about how to let text-generation-webui use an already running standalone alltalk_tts via alltalk_tts api's.

### DISCUSSION THREAD

**Discussion by billtown** [erew123/alltalk_tts#167] [General]
I think I'm just being dense here, but, is there an option to use this as an extension, without installing tons of cruft in the extension?  I feel as though I should just be able to configure confignew.json with the needed options to point to an already running container, however, the seemingly needed interactive installation, of running atsetup.sh installs a ton of requirements, and launches alltalk_tts via launch.sh once configured, which also downloads models once the plugin is enabled.... is this needed or api access?

Once the plugin is running successfully, I can see it making calls to my already running alltalk docker, however, it's always looking locally for the output and not using the api.  My best recollection is I'm supposed to use "API Local" as the settings in text-generation-webui, however, I'm just not clear on what the smallest needed requirements are, in order for the text-generation-webui plugin portion of code to use a running alltalk_tts's api.  It's also a little weird that it attempts to "reload" by hitting the api, and returns a 500, when my container is actually returning a 405.

I have some sort of disconnect on what the methodology is supposed to be here :)  I just want a plugin that can use alltalk_tts's api, that works with text-generation-webui and I'm guessing I have the most bloated method of doing that :)

It's like it's using the api's, but then also not actually using them and I'm confused trying to reverse figure out the methodology here :)  

Can the text-generation-webui plugin use a standalone Alltalk_tts server?

Great project, and ty for it :)

**Comment by erew123** [comment]
Hi @billtown 

Im a little lost figuring what you're doing here, though I am also deep in code right at this moment, so forgive my mental state to take in lots of other information. 

So you aren't running Text-gen-webui on your local machine on your local disk? If you are running it on your local hard drive (not in a docker) you would just follow this installation https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-quick-setup-text-generation-webui--standalone-installation and as shown on this video https://www.youtube.com/watch?v=icn2XS5rUH8 which would install AllTalk in Text-generation-webui's python environment and install any missing Python requirements. 

If on the other hand you are running the AllTalk docker, I'm not sure if you read my statement on docker builds on the front page? https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-docker-builds-and-google-colabs 

The docker itself is a standalone thing and will have no integration with Text-gen-webui.

Docker is something I am working to try clean up as much as possible, though that's all part of a much larger update at this time. 

**So, I think if I am reading your question correctly**, you would like to be able to run AllTalk on computer system A and Text-gen-webui on computer system B, with the bit that is running on Text-gen-webui, just being the interface portion and communicating over the network to the AllTalk server running on system A....... is that correct? If it is, it doesnt do that (currently).... can I make it do that... yeah I should be able to.... However, the confignew.json file is used between both systems currently as they both need to share certain information e.g. Text-gen-webui's interface would need to know about the remote system and if DeepSpeed is available (for example), things like that are possible through the api, which does now carry some information like that..... but Ive not worked on Text-gen-webui's interface/integration for a while now. Its potentially something I can add in future. Im currently working towards an update of various things with the API, which would make it simpler to do in future.

Thanks

**Reply by billtown** [reply]
aha! yeah I think that's why I was confused :)  ty.  I've not worked with text-generation-webui's extension stuff at all (I've not written an extension), so I was just trying to figure out what was supposed to be happening :)  And yes, my preference is to run components/services separately, So maybe I can take a shot at utilizing the api or mangle a plugin to do so.  I saw your changes coming :) Thank you for clarifying. Apologies for my confusion :)

**Comment by erew123** [comment]
Well, I'm ? away in days from completing the new API. It is a question of time and testing testing testing + documentation before I send it live, but in theory, it should make the scenario you want a pretty simple one to do, without too many problems. I mean, I have it working, but Im trying to get quite a few other bits all done at one go, in one big push.

Re messing in the code as it stands currently, Im not so sure it would be super easy, but I cant recall all the code off the top of my head atm. I think it pulls back a local disk version of the wav files in text-gen rather than a network request, hence over the network isnt a simple change.

Ill make a note of your request though and if everything goes well, Im reasonably sure I can figure something like that in, either at time of new API release or not too much after (pending all going well). Hopefully I should be able to release a google colab and probably a new docker at the same time. 

Ill close the ticket off for now and I guess watch this space!

Thanks
